### ISSUE TITLE
Behavior in two-level navigation menu

### ISSUE THREAD

**Issue by odi76** [bsgriggs/mendix-unsaved-changes-message#2]
Hi Benjamin,

Another issue. I use a Navigation Tree having two level items. I applied the both CCS classes *(unsaved-changes-block and mx-navigationtree).

Though I experience selection action is caught, it is also caught on the menu items of level 1. This results in the following scenario:
1) User selects an L1 item to expand the subitems for navigation.
2) Application asks the confirmation for proceeding.
3) User accepts.
4) Application presents L2 items under L1.
5) User selects an L2 item to navigate.
6) Application asks user to confirm proceeding again.
7) Users accepts again.
8) Application navigates user to the page associated to the menu item.

This duplicated confirmation should be reduced to only 1 which really triggers leaving the page. Do you have any idea?

Thanks in advance!

br,
Istvan

**Comment by bsgriggs** [comment]
Hi Istvan,
Unfortunately, this widget does not work with multi-level navigation menus. There's no was to target the sub-menus as they are closed when the page loads and the widget adds the blockers to the page.

**Comment by bsgriggs** [comment]
Hi Istvan,
I've been wrestling with a few changes that could fix this issue. I've released v2.0.0, the code now looks for the items and sub-menu items instead of the menu as a whole. BUT it will cause strange behavior if your nav menu is huge and doesn't work well with hover states.

The widget in v2.0.0 has a new mode 'JavaScript Action'. I will be releasing a module in the coming week(s) that has this JavaScript Action and some examples or you can use this attached [file](https://github.com/user-attachments/files/18832130/CheckUnsavedChangesMessage.mpk.zip) that the module will have. 

Basically for the most robust/consistant UX, you would need to have **_all_** actions that can leave the page (including the nav menu actions) call a Nanoflow with this JavaScript action. More details in the [ReadMe](https://github.com/bsgriggs/mendix-unsaved-changes-message?tab=readme-ov-file#observe-mode---mendix--both---javascript-action). This will _absolutely_ fix your problem, but may be an effort if you have a huge menu. This method also has the added benefit of working with keyboard controls.

-Benjamin
